src/main/scala/dpla/ingestion3/CdlHarvesterMain.scala

```diff
@@ -32,23 +32,37 @@ object CdlHarvesterMain extends ApiHarvester {
     */
   def main(args: Array[String]): Unit = {
     // Parse and set the command line options
-    val conf = new CdlHarvestConf(args)
-    val outFile = new File(conf.outputFile.getOrElse("out"))
-    val queryParams = Map(
-      "query" -> conf.query.getOrElse("*:*"),
-      "rows" -> conf.rows.getOrElse("100"),
-      "api_key" -> conf.apiKey.getOrElse("MISSING API KEY")
-    )
+    val cmdArgs = new HarvestCmdArgs(args)
 
+    val outputDir = cmdArgs.output.toOption match {
```

Comment:
Rather than using a pattern match, you can call `.map` on an option:

`cmdArgs.output.toOption.map(new File(_)).getOrElse(throw new RuntimeException("No output specified")`

Reply:
😎